### PR TITLE
feat: rename command

### PR DESCRIPTION
**Pull request by jarulraj**
1. added support for rename operator
2. updated gitignore and added changelog
3. updates to command prompt
4. refactored code in pytorch_abstract_udf etc. to increase coverage

### CODE REVIEW

File: eva/catalog/catalog_manager.py [hunk 1]
```diff
@@ -249,6 +249,10 @@ def drop_udf(self, udf_name: str) -> bool:
         """
         return self._udf_service.drop_udf_by_name(udf_name)
 
+    def rename_table(self, new_name: str, metadata_id: int):
```
Comment (gaurav274) [inline eva/catalog/catalog_manager.py:252]: We have stopped working with ids and rather pass the catalog object

File: eva/catalog/services/df_service.py [hunk 2]
```diff
@@ -113,3 +113,16 @@ def drop_dataset_by_name(self,
                 "Delete dataset failed for name {}".format(dataset_name))
             return False
         return True
+
+    def rename_dataset_by_id(self, new_name: str, metadata_id: int) -> bool:
```
Comment (gaurav274) [inline eva/catalog/services/df_service.py:117]: Same no ids

File: eva/optimizer/operators.py [hunk 3]
```diff
@@ -442,6 +443,51 @@ def __hash__(self) -> int:
...
+class LogicalRename(Operator):
+    """Logical node for rename table operations
+
+    Arguments:
+        old_table {TableRef}: [old table that is to be renamed]
+        catalog_table_id {int}: [catalog id for the old table]
+        new_name {TableInfo}: [new name for the old table]
+    """
+
+    def __init__(self, old_table_ref: TableRef,
```
Comment (gaurav274) [inline eva/optimizer/operators.py:455]: catalog_id not required here

File: eva/optimizer/optimizer_utils.py [hunk 4]
```diff
@@ -74,3 +75,17 @@ def extract_equi_join_keys(join_predicate: AbstractExpression,
                     right_join_keys.append(left_child)
 
     return (left_join_keys, right_join_keys)
+
+
+def bind_table_ref(video_info: TableInfo) -> int:
```
Comment (gaurav274) [inline eva/optimizer/optimizer_utils.py:80]: We don't. need this logic anymore. Handled by the binder now

File: eva/optimizer/rules/parser_visitor/_drop_statement.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from eva.parser.evaql.evaql_parserVisitor import evaql_parserVisitor
+from eva.parser.table_ref import TableRef
+from eva.parser.drop_statement import DropTableStatement
+from eva.parser.evaql.evaql_parser import evaql_parser
+
+
+class DropTable(evaql_parserVisitor):
```
Comment (gaurav274) [inline eva/optimizer/rules/parser_visitor/_drop_statement.py:22]: This file should already be there

File: eva/optimizer/rules/parser_visitor/_rename_statement.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+from eva.utils.logging_manager import logger
+
+# Modified
+##################################################################
+# RENAME STATEMENTS
+##################################################################
+
+
+class RenameTable(evaql_parserVisitor):
+    def visitOldtableName(self, ctx: evaql_parser.OldtableNameContext):
```
Comment (gaurav274) [inline eva/optimizer/rules/parser_visitor/_rename_statement.py:30]: We don't need new constructs for newTable and oldTable. Rather we can just use the tableName for both

File: eva/optimizer/rules/rules.py [hunk 7]
```diff
@@ -405,6 +410,24 @@ def apply(self, before: LogicalCreate, context: OptimizerContext):
...
+        # pattern.append_child(Pattern(OperatorType.DUMMY))
+        super().__init__(RuleType.LOGICAL_RENAME_TO_PHYSICAL, pattern)
+
+    def promise(self):
+        return Promise.LOGICAL_RENAME_TO_PHYSICAL
+
+    def check(self, before: Operator, context: OptimizerContext):
+        return True
+
+    def apply(self, before: LogicalRename, context: OptimizerContext):
```
Comment (gaurav274) [inline eva/optimizer/rules/rules.py:425]: Get rid of id

File: eva/optimizer/rules/rules.py [hunk 8]
```diff
@@ -773,6 +796,7 @@ def apply(self, before: LogicalProject, context: OptimizerContext):
 ##############################################
 
 
+# Modified
```
Comment (gaurav274) [inline eva/optimizer/rules/rules.py:799]: Remove comment

File: eva/optimizer/statement_to_opr_convertor.py [hunk 9]
```diff
@@ -194,6 +197,19 @@ def visit_create(self, statement: AbstractStatement):
             table_ref, statement.column_list, statement.if_not_exists)
         self._plan = create_opr
 
+    def visit_rename(self, statement: RenameTableStatement):
```
Comment (gaurav274) [inline eva/optimizer/statement_to_opr_convertor.py:200]: Some logic will be handled by binder

File: eva/optimizer/rules/parser_visitor/_drop_statement.py (hunk 5 above)
Comment (gaurav274) [reply eva/optimizer/rules/parser_visitor/_drop_statement.py:22]: Delete this file. There shouldn't be one inside the rules

File: eva/optimizer/rules/parser_visitor/_rename_statement.py (hunk 6 above)
Comment (gaurav274) [reply eva/optimizer/rules/parser_visitor/_rename_statement.py:30]: Delete this file

File: eva/parser/types.py [hunk 10]
```diff
@@ -22,13 +22,15 @@ class ColumnConstraintEnum(Enum):
     UNIQUE = auto()
 
 
+# Modified, add RENAME, add TRUNCATE
```
Comment (gaurav274) [inline eva/parser/types.py:25]: Remove this

File: eva/planner/rename_plan.py [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+
+# Modified
+class RenamePlan(AbstractPlan):
+    """
+    This plan is used for storing information required for rename table
+    operations.
+    Arguments:
+        old_table {TableRef} -- table ref for table to be renamed in storage
+        table_id {int} -- catalog table id for the table
```
Comment (gaurav274) [inline eva/planner/rename_plan.py:27]: id not needed

File: eva/udfs/pytorch_abstract_udf.py [hunk 12]
```diff
@@ -23,18 +23,9 @@
...
-except ImportError as e:
-    raise ImportError(f"Failed to import with error {e}, \
-        please try `pip install torch`")
-
-try:
-    from torchvision.transforms import Compose, transforms
-except ImportError as e:
-    raise ImportError(f"Failed to import with error {e}, \
-        please try `pip install torchvision`")
+import torch
```
Comment (gaurav274) [inline eva/udfs/pytorch_abstract_udf.py:26]: Rather add test case for this

File: eva/catalog/catalog_manager.py (hunk 1 above)
Comment (jarulraj) [reply eva/catalog/catalog_manager.py:252]: This was left by the students who added support of rename. Please fix it to use the current approach.

File: eva/udfs/pytorch_abstract_udf.py (hunk 12 above)
Comment (jarulraj) [reply eva/udfs/pytorch_abstract_udf.py:26]: have we decided to make `torch` optional?

Comment (gaurav274) [reply eva/udfs/pytorch_abstract_udf.py:26]: @xzdandy @pchunduri6 What do you guys think?

Comment (xzdandy) [reply eva/udfs/pytorch_abstract_udf.py:26]: Right now. `pip` uses fixed version of dependency (https://github.com/georgia-tech-db/eva/blob/master/setup.cfg). Will this become a problem when users install their own torch version?

Comment (pchunduri6) [reply eva/udfs/pytorch_abstract_udf.py:26]: I actually removed the torch installation from setup.cfg and had to install it separately anyway for ensuring compatible cuda version.

File: CHANGELOG.md [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [0.0.3] - 2022-07-03 
+### [Breaking Changes]
+
+### [Added]
+
+### [Changed]
+* PR 263 : Updates to logger for more informative messages
+* PR 261 : Improving clarity in README.md
```
Comment (xzdandy) [inline CHANGELOG.md:14]: Do we need to update the CHANGELOG?

File: eva/catalog/services/df_service.py [hunk 14]
```diff
@@ -93,23 +93,39 @@ def dataset_object_by_name(self, database_name, dataset_name,
...
-        except Exception:
-            logger.error(
-                "Delete dataset failed for name {}".format(dataset_name))
-            return False
-        return True
+        except Exception as e:
+            err_msg = "Delete dataset failed for name {} with error {}".format(
+                dataset_name, str(e)
+            )
+            logger.error(err_msg)
```
Comment (xzdandy) [inline eva/catalog/services/df_service.py:114]: Shall we use `logger.exception`? So it prints the full stack trace in the server log.

File: eva/server/interpreter.py [hunk 15]
```diff
@@ -16,13 +16,29 @@
 from cmd import Cmd
 from contextlib import ExitStack
 from eva.server.db_api import connect
+from os import path
+from readline import read_history_file, write_history_file, set_history_length
+
+# History file to persist EVA  command history across multiple client sessions
+histfile = 'eva.history'
+histfile_size = 1000
```
Comment (xzdandy) [inline eva/server/interpreter.py:25]: Should this be configurable in eva.yml? histfile is `~/.eva/eva.history`?

File: eva/version.py [hunk 16]
```diff
@@ -0,0 +1,7 @@
+_MAJOR = "0"
+_MINOR = "0"
+_REVISION = "3"
+
```
Comment (xzdandy) [inline eva/version.py:4]: Do we need update this every time manually?

**Review by xzdandy — APPROVED** [review]
Overall LGTM. Leave some comments.

**Pull request merged**